dynatrace/environment_v2/extensions.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright 2021 Dynatrace LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from datetime import datetime
+from dynatrace.dynatrace_object import DynatraceObject
+from typing import List, Optional, Union, Dict, Any
+
+from dynatrace.http_client import HttpClient
+from dynatrace.pagination import PaginatedList
+
+
+
+class ExtensionsServiceV2:
+    ENDPOINT = "/api/v2/extensions"
+    SCHEMA_ENDPOINT = "/api/v2/extensions/schemas"
+
+
+    def __init__(self, http_client: HttpClient):
+        self.__http_client = http_client
+
+
+    def list(self) -> PaginatedList["MinimalExtension"]:
+        """ Lists all the extensions 2.0 available in your environment
+
+        :return: a list of MinimalExtension objects
+        """
+        return PaginatedList(MinimalExtension, self.__http_client, target_url=self.ENDPOINT, list_item="extensions")
+
+
+    def listByName(self, extension_name: str) -> PaginatedList["MinimalExtension"]:
+        """ Lists all the extensions 2.0 by specified name in your environment
+
+        :param extension_name: the name of the extension 2.0
+
+        :return: a list of MinimalExtension objects
+        """
+        return PaginatedList(MinimalExtension, self.__http_client, target_url=f"{self.ENDPOINT}/{extension_name}", list_item="extensions")
+
+
+    def listEvents(self, extension_name: str) -> PaginatedList["ExtensionEventDTO"]:
+        """ List of the latest extension environment configuration events
+
+        :param extension_name: the name of the extension 2.0
+
+        :return: a list of ExtensionEventDTO object
+        """
+        return PaginatedList(ExtensionEventDTO, self.__http_client, target_url=f"{self.ENDPOINT}/{extension_name}/environmentConfiguration/events", list_item="extensionEvents")
+
+
+    def listEventsByConfigId(self, extension_name: str, config_id: str) -> PaginatedList["ExtensionEventDTO"]:
```

Comment:
Call this `list_monitoring_config_events`, and same thing as before, `list_monitoring_config_events(config_id)` method on the `MinimalExtension` instance